reactnative-todo/ios/reactnativetodo/FRAuthSampleStructs.swift

```diff
@@ -0,0 +1,250 @@
+/**
+ * FRAuthSampleBridge.m
+ * reactnativetodo
+ *
+ * Copyright (c) 2021 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+import Foundation
+import FRAuth
+
+/**
+ * The `node` or "step" resulting from a continuation of a journey/tree
+ */
+public struct FRNode: Encodable {
+  var frCallbacks: [FRCallback]
+  /// ID for the journey/tree progress
+  var authId: String
+  /// Unique UUID String value of initiated AuthService flow
+  var authServiceId: String
+  /// Stage attribute in Page Node
+  var stage: String?
+  /// Header attribute in Page Node
+  var pageHeader: String?
+  /// Description attribute in Page Node
+  var pageDescription: String?
+  //array of raw callbacks
+  var callbacks: [[String: Any]]
+
+  private enum CodingKeys: String, CodingKey {
+    case frCallbacks, authId, authServiceId, stage, pageHeader, pageDescription
+  }
+
+  init(node: Node) {
+    authId = node.authId
+    authServiceId = node.authServiceId
+    stage = node.stage
+    pageHeader = node.pageHeader
+    pageDescription = node.pageDescription
+    frCallbacks = [FRCallback]()
+    callbacks = [[String: Any]]()
+
+    for callback in node.callbacks {
+      callbacks.append(callback.response)
+      frCallbacks.append(FRCallback(callback: callback))
+    }
+  }
+
+  /**
+   * Method for passing the building the `node` object and serializing it for the React Native layer
+   */
+  func resolve() throws -> String  {
+    var response = [String: Any]()
+    response["authId"] = self.authId
+    response["authServiceId"] = self.authServiceId
+    response["stage"] = self.stage
+    response["description"] = self.pageDescription
+    response["header"] = self.pageHeader
+    response["callbacks"] = self.callbacks
+    return try response.toJson()
+  }
+}
+
+/**
+ * The `node.callbacks` item
+ */
+public struct FRCallback: Encodable {
+  var type: String
+  var prompt: String?
+  var choices: [String]?
+  var predefinedQuestions: [String]?
+  var inputNames: [String]?
+  var policies: RawPolicies?
+  var failedPolicies: [RawFailedPolicies]?
+
+  /**
+   * Raw journey/tree JSON response from ForgeRock server
+   */
+  var response: String
+
+  init(callback: Callback) {
+    self.type = callback.type
+
+    if let thisCallback = callback as? SingleValueCallback {
+      self.prompt = thisCallback.prompt
+      self.inputNames = [thisCallback.inputName!]
```

Comment:
Is it always guaranteed for the callback inputName have a value? I'm a little afraid of using forced unwrapping. Maybe we could have provided a default value instead:   `self.inputNames = [thisCallback.inputName ?? "default_value"]

Reply:
Good catch this has been missed, it shouldn't be like that and will be changed